packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1330,4 +1333,253 @@ void main() {
       textDirection: TextDirection.ltr,
     ));
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/6751
+  testWidgets('When InkWell/Ancestor has a GlobalKey and changes position, splash should not stop', (WidgetTester tester) async {
+    void expectTest(bool painted) {
+      final PaintPattern paintPattern = paints..circle();
+
+      expect(
+        Material.of(tester.element(find.byType(InkWell)))! as RenderBox,
+        painted ? paintPattern : isNot(paintPattern),
+      );
+    }
+
+    Future<void> test(Widget Function(Key? key, {bool onTapDownChangeWrap, bool onTapChangeWrap}) buildApp) async {
+      await tester.pumpWidget(buildApp(null, onTapChangeWrap: true));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(false);
+      await tester.pumpAndSettle();
+      expectTest(false);
+
+      await tester.pumpWidget(buildApp(const Key('foo'), onTapChangeWrap: true));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(false);
+      await tester.pumpAndSettle();
+      expectTest(false);
+
+      // Does not call setState.
+      await tester.pumpWidget(buildApp(GlobalKey()));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(true);
```

Comment:
i don't understand why this test fails with line 1359 set to false and line 1367 set to true. What's the difference between those two cases?

Reply:
oh, i see, in the cases above the ink well is reparented so fast that the first frame has it moved already.

Reply:
i think the test would be more useful with checks immediately after tap (pump no duration), then after a short duration but before the move, then after the move, then at the end (after "settle" or the big pump).